### PR TITLE
Feature/GitHub tests

### PR DESCRIPTION
**Pull request by JVickery-TBS**
- Modified the github tests workflow to function with our ckan 2.9 setup (with python2.7 and not 3)
- Removed `pylons` usages throughout the code
- Switch all the tests from using `FunctionalTestBase` to now using `pytest` with `@pytest.mark.usefixtures('clean_db')`
- Changed test classes `setup` methods to just `__init__` as they are no longer children of `FunctionalTestBase`
- Removed the `logged_in` method from the user controller and replaced it with a Flask View override of the user `login` and `logged_in` view functions.

TODO: replace all of the `nose` assertion usages with basic asserts.

### CODE REVIEW

**Comment by JVickery-TBS** [comment]
@wardi well, apparently, I did it. 

I have some TODOs in there to assert the raises to expected error messages. Let me know if you want me to do that as part of this PR or later, as it used to just use `nose.tools.assert_raises` which just checked if the code raised the given exception class.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # ckanext-canada
 
-[![Tests](https://github.com/open-data/ckanext-canada/workflows/Tests/badge.svg?branch=master)](https://github.com/open-data/ckanext-canada/actions?query=name%3ATests)
+[![Tests](https://github.com/open-data/ckanext-canada/workflows/Tests/badge.svg?branch=canada-v2.9&event=push)](https://github.com/open-data/ckanext-canada/actions?query=name%3ATests)
```
Comment (wardi) [inline README.md:3]: looks like something we'll forget to change :thinking:

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -17,13 +17,13 @@ jobs:
...
       matrix:
-        ckan-version: [2.8]
+        ckan-version: [2.9]
       fail-fast: false
 
     name: CKAN ${{ matrix.ckan-version }}
     runs-on: ubuntu-20.04
     container:
-      image: openknowledge/ckan-dev:${{ matrix.ckan-version }}
+      image: openknowledge/ckan-dev:2.8
```
Comment (wardi) [inline .github/workflows/test.yml:26]: add a comment here about why we need to use the 2.8 version of the image?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -17,13 +17,13 @@ jobs:
     # needs: lint
     strategy:
       matrix:
-        ckan-version: [2.8]
+        ckan-version: [2.9]
```
Comment (wardi) [inline .github/workflows/test.yml:20]: version is hard-coded in a number of places, should we remove this setting?

File: ckanext/canada/tests/fixtures.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+from ckanapi import LocalCKAN
+from factory import Sequence
+from ckanext.canada.tests.factories import CanadaOrganization as Organization
+
+import logging
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def prepare_dataset_type_with_resources(request):
```
Comment (wardi) [inline ckanext/canada/tests/fixtures.py:13]:
```suggestion
def prepare_recombinant_dataset_type_with_resources(request):
```

File: ckanext/canada/tests/test_briefingt.py [hunk 5]
```diff
@@ -1,31 +1,33 @@
 # -*- coding: UTF-8 -*-
-from nose.tools import assert_equal, assert_raises
 from ckanapi import LocalCKAN, ValidationError
 
-from ckan.tests.helpers import FunctionalTestBase
-from ckanext.canada.tests.factories import CanadaOrganization as Organization
+import pytest
+from ckanext.canada.tests.fixtures import prepare_dataset_type_with_resources
```
Comment (wardi) [inline ckanext/canada/tests/test_briefingt.py:5]: do you need this import? I thought pytest had some magic to collect fixtures

Comment (wardi) [reply ckanext/canada/tests/test_briefingt.py:5]: (like the clean_db one used below)

File: ckanext/canada/tests/test_pd.py [hunk 6]
```diff
@@ -1,37 +1,32 @@
...
+
 
     def test_top_bucket(self):
-        assert_equal(
-            dollar_range_facet('foo', [100, 500, 1000], 5000), {
-                'foo_range': u'2',
-                'foo_en': u'A: $1,000.00+',
-                'foo_fr': u'A: 1\xa0000,00\xa0$ +'})
+        assert dollar_range_facet('foo', [100, 500, 1000], 5000) == {
+           'foo_range': u'2',
```
Comment (wardi) [inline ckanext/canada/tests/test_pd.py:16]:
```suggestion
            'foo_range': u'2',
```

**Comment by wardi** [comment]
that was a lot, and this method of passing parameters to fixtures isn't pretty, but tests are green :-)

File: .github/workflows/test.yml (hunk 3 above)
Comment (JVickery-TBS) [reply .github/workflows/test.yml]: Removed matrix stuff.

File: ckanext/canada/tests/fixtures.py (hunk 4 above)
Comment (JVickery-TBS) [reply ckanext/canada/tests/fixtures.py]: n/a removed custom fixtures

File: ckanext/canada/tests/test_briefingt.py (hunk 5 above)
Comment (JVickery-TBS) [reply ckanext/canada/tests/test_briefingt.py]: n/a removed custom fixtures

File: ckanext/canada/tests/test_pd.py (hunk 6 above)
Comment (JVickery-TBS) [reply ckanext/canada/tests/test_pd.py]: fixed.

File: .github/workflows/test.yml (hunk 2 above)
Comment (JVickery-TBS) [reply .github/workflows/test.yml:22]: Comment added.

File: README.md (hunk 1 above)
Comment (JVickery-TBS) [reply README.md]: Changed to master branch. People will get emails for tests failing in canada-v2.9 anyways, so should not need visual badge for this specific branch.

**Comment by JVickery-TBS** [comment]
@wardi I have cleaned everything up. Namely, instead of the custom strange fixtures, I am now only using the setup class methods and directly calling `reset_db`.

So it all looks a lot nicer now and makes way more sense.

Other comments have been addressed.

As discussed in slack, the `test_example` and `test_multiple_errors` in the consultations tests do not work in that order. Which is currently a mystery as the other tests do very similar things with `datastore_upsert`. I will try your suggestion of putting `None` as the values to the expected erroring fields.

File: ckanext/canada/tests/test_ati.py [hunk 7]
```diff
@@ -1,55 +1,97 @@
...
+        """
+        reset_db()
+
+        log.info('Running setup for {}'.format(self.__name__))
+
         org = Organization()
         lc = LocalCKAN()
+
+        log.info('Creating organization with id: {}'.format(org['name']))
+        log.info('Setting organization dataset type to {}'.format('ati'))
```
Comment (wardi) [inline ckanext/canada/tests/test_ati.py:28]:
```suggestion

        org = Organization()
        lc = LocalCKAN()

```

Comment (wardi) [reply ckanext/canada/tests/test_ati.py:28]: debug logging probably not required here

Comment (JVickery-TBS) [reply ckanext/canada/tests/test_ati.py]: Logging removed.

**Comment by JVickery-TBS** [comment]
@wardi Okay I think this is good to go now.

File: ckanext/canada/tests/test_consultations.py [hunk 8]
```diff
@@ -82,14 +82,6 @@ def test_multiple_errors(self):
...
 
 
-    def test_example(self):
-        lc = LocalCKAN()
-        record = get_chromo('consultations')['examples']['record']
-        lc.action.datastore_upsert(
-            resource_id=self.resource_id,
-            records=[record])
-
-
```
Comment (wardi) [inline ckanext/canada/tests/test_consultations.py:67]: just to confirm, moving this method above isn't required for the tests to pass now, right?

Comment (JVickery-TBS) [reply ckanext/canada/tests/test_consultations.py:67]: @wardi Correct, the order of the methods does not matter now with the change to `setup_method`

Comment (wardi) [reply ckanext/canada/tests/test_consultations.py:67]: ok, then this all looks good

**Pull request merged**